src/main/java/shef/data/Recipe.java

```diff
@@ -62,12 +66,19 @@ public Recipe(String name, String description, Set<String> tags, Set<String> ing
     this.timestamp = timestamp;
   }
 
-  /** Creates a Recipe from a Datastore entity. */
+  /** 
+   * Creates a Recipe from a Datastore entity.
+   * This constructor is used to easily convert an Entity into an object, which can be sent as JSON.
+   */
   public Recipe(Entity recipeEntity) {
     this.name = (String) recipeEntity.getProperty("name");
     this.description = (String) recipeEntity.getProperty("description");
+    this.time = (double) recipeEntity.getProperty("time");
```

Comment:
do any of the other constructors also need these fields?

For the future: we should have used constructor chaining here, but don't worry about it now.